src/alley/wp/validator/class-block-innerblocks-count.php

```diff
@@ -0,0 +1,248 @@
+<?php
+/**
+ * Block_InnerBlocks_Count class file
+ *
+ * (c) Alley <info@alley.co>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @package wp-match-blocks
+ */
+
+namespace Alley\WP\Validator;
+
+use Alley\Validator\AlwaysValid;
+use Alley\Validator\Comparison;
+use Alley\Validator\Not;
+use Alley\Validator\WithMessage;
+use Laminas\Validator\Exception\InvalidArgumentException;
+use Laminas\Validator\ValidatorInterface;
+use Traversable;
+use WP_Block;
+use WP_Block_Parser_Block;
+use WP_Error;
+
+/**
+ * Validates whether the given block has a number of inner blocks.
+ */
+final class Block_InnerBlocks_Count extends Block_Validator {
+	/**
+	 * Array of validation failure message templates.
+	 *
+	 * @var string[]
+	 */
+	protected $messageTemplates = [
+		'not_equal'                    => '',
+		'not_identical'                => '',
+		'is_equal'                     => '',
+		'is_identical'                 => '',
+		'not_less_than'                => '',
+		'not_greater_than'             => '',
+		'not_less_than_or_equal_to'    => '',
+		'not_greater_than_or_equal_to' => '',
+		'invalid_comparison'           => '',
+		'default'                      => '',
+	];
+
+	/**
+	 * Array of additional variables available for validation failure messages.
+	 *
+	 * @var string[]
+	 */
+	protected $messageVariables = [
+		'count' => [
+			'options' => 'count',
+		],
+	];
+
+	/**
+	 * Options for this validator.
+	 *
+	 * @var array
+	 */
+	protected $options = [
+		'operator' => '>=',
+		'count'    => 0,
+	];
+
+	/**
+	 * Valid inner block counts based on options.
+	 *
+	 * @var ValidatorInterface
+	 */
+	private ValidatorInterface $valid_comparisons;
+
+	/**
+	 * Set up.
+	 *
+	 * @param array|Traversable $options Validator options.
+	 */
+	public function __construct( $options = null ) {
+		$this->localize_templates();
+
+		$this->valid_comparisons = new Comparison(
+			[
+				'operator' => $this->options['operator'],
+				'compared' => $this->options['count'],
+			],
+		);
+
+		parent::__construct( $options );
+	}
+
+	/**
+	 * Sets one or multiple options. Refreshes the cached validators for the comparisons.
+	 *
+	 * @throws InvalidArgumentException If requested comparisons are invalid.
+	 *
+	 * @param array|Traversable $options Options to set.
+	 * @return self
+	 */
+	public function setOptions( $options = [] ) {
+		parent::setOptions( $options );
+
+		try {
+			$this->valid_comparisons = new Comparison(
```

Comment:
This is a repeat of what is in the constructor. Could the constructor call this function instead of setting `new Comparison` directly?